tardis/visualization/tools/convergence_plot.py

```diff
@@ -0,0 +1,440 @@
+        fig = fig.update_layout(
+            xaxis=dict(range=[0, self.iterations + 1], dtick=2),
+            xaxis2=dict(
+                matches="x",
+                range=[0, self.iterations + 1],
+                dtick=2,
+            ),
+            xaxis3=dict(
+                title=r"$\mbox{Iteration Number}$",
+                dtick=2,
+            ),
+            yaxis=dict(
+                title=r"$T_{\text{inner}}\ [\text{K}]$",
+                automargin=True,
+                tickformat="g",
+                exponentformat="e",
+                nticks=4,
+            ),
+            yaxis2=dict(
+                exponentformat="e",
+                title=r"$\text{Luminosity}~[\text{erg s}^{-1}]$",
+                title_font_size=13,
+                automargin=True,
+                nticks=7,
+            ),
+            yaxis3=dict(
+                title=r"$~~\text{Residual}\\\text{Luminosity[%]}$",
+                title_font_size=12,
+                automargin=True,
+                nticks=4,
+            ),
+            height=630,
+            hoverlabel_align="right",
+            margin=dict(
+                b=25, t=25, pad=0
+            ),  # reduces whitespace surrounding the plot
+        )
+
+        # allow overriding default layout
+        if hasattr(self, "t_inner_luminosities_config"):
+            self.override_plot_parameters(fig, self.t_inner_luminosities_config)
+
+        self.t_inner_luminosities_plot = fig
+
+    def override_plot_parameters(self, fig, parameters):
+        """
+        Override default plot properties.
+
+        Any property inside the data dictionary is however, applied equally across all traces.
+        This means trace-specific data properties can't be changed using this function.
+
+        Parameters
+        ----------
+        fig : go.FigureWidget
+            FigureWidget object to be updated
+        parameters : dict
+            Dictionary used to update the default plot style.
+        """
+        # because fig.data is a tuple of traces, a property in the data dictionary is applied to all traces
+        # the fig is a nested dictionary, any property n levels deep is not changed until the value is a not dictionary
+        # fig["property_1"]["property_2"]...["property_n"] = "value"
+        for key, value in parameters.items():
+            if key == "data":
+                # all traces will have same data property
+                for trace in list(fig.data):
+                    self.override_plot_parameters(trace, value)
+            else:
+                if type(value) == dict:
+                    self.override_plot_parameters(fig[key], value)
+                else:
+                    fig[key] = value
+
+    def build(self, display_plot=True):
+        """
+        Create empty convergence plots and display them.
+
+        Parameters
+        ----------
+        display_plot : bool, optional
+            Displays empty plots. Defaults to True.
```

Comment:
`default: True` should go in type instead of description - like you did everwhere else